### PR TITLE
Updating What's New for 2.3.0

### PR DESCRIPTION
**Pull request by gcolliso**

### CODE REVIEW

**Comment by gcolliso** [comment]
- Should we list `statsz` in [monitoring](https://docs.nats.io/nats-server/configuration/monitoring#monitoring-endpoints)? How is this different from the endpoint `jsz`?
- Do we have a reference list we can link to of the JetStream errors?

File: whats_new.md [hunk 1]
```diff
@@ -2,6 +2,22 @@
...
 The NATS.io team is always working to bring you features to improve your NATS experience. Below you will find feature summaries for new implementations to NATS. Check back often for release highlights and updates.
 
+## Server release v2.3.0
+
+### JetStream
+* Richer API errors. JetStream errors now contain an ErrCode that uniquely describes the error. 
+* Ability to send more advanced Stream purge requests
+* Stream can now be configured with a per-subject message limit
+* Encryption at rest
+### Monitoring
```
Comment (ripienaar) [inline whats_new.md:12]: space above, I guess

File: whats_new.md [hunk 2]
```diff
@@ -2,6 +2,22 @@
 
 The NATS.io team is always working to bring you features to improve your NATS experience. Below you will find feature summaries for new implementations to NATS. Check back often for release highlights and updates.
 
+## Server release v2.3.0
+
+### JetStream
+* Richer API errors. JetStream errors now contain an ErrCode that uniquely describes the error. 
+* Ability to send more advanced Stream purge requests
```
Comment (ripienaar) [inline whats_new.md:9]: ....that can purge all messages for a specific subject

File: whats_new.md [hunk 3]
```diff
@@ -2,6 +2,22 @@
...
+* Stream can now be configured with a per-subject message limit
+* Encryption at rest
```
Comment (ripienaar) [inline whats_new.md:11]: Encryption of JetStream data at rest

**Review by ripienaar — CHANGES_REQUESTED** [review]
minor things tbh, fine as is too

**Review by kozlovic — APPROVED** [review]
LGTM with RI corrections.

**Comment by gcolliso** [comment]
Sorry I wasn't clear @ripienaar & @kozlovic  - I put questions in the comments that I need help addressing.

- Should we list `statsz` in [monitoring](https://docs.nats.io/nats-server/configuration/monitoring#monitoring-endpoints)? How is this different from the endpoint `jsz`?
- Do we have a reference list we can link to of the JetStream errors?

**Comment by ripienaar** [comment]
> Sorry I wasn't clear @ripienaar & @kozlovic - I put questions in the comments that I need help addressing.
> 
>     * Should we list `statsz` in [monitoring](https://docs.nats.io/nats-server/configuration/monitoring#monitoring-endpoints)? How is this different from the endpoint `jsz`?

it's kind of a minor feature in things users dont tend to touch, I wouldnt mention it

>     * Do we have a reference list we can link to of the JetStream errors?

Not exactly at the moment now, there's some new CLI tooling to lookup error codes but thats all for now, kind of emerging feature.

**Comment by kozlovic** [comment]
Regarding `statsz` I guess this is my mistake in release notes to have added that under monitoring. This is not a monitoring endpoint per-se, but is available through system events (which normally mimic what one can get from monitoring endpoints). So it is still monitoring, but not one you get from monitoring webpage.

**Pull request merged**